### ISSUE TITLE
Having trouble defining my actions in Ember

### ISSUE THREAD

**Issue by novakw267** [ga-wdi-boston/capstone-project#549]
I am having a bit of trouble wrapping my head around the structure of ember and how each part talks to each other. The main problem I am attempting to fix at the moment is 

```

message
:
"Assertion Failed: Attempted to register a view with an id already in use: next"
``` 
 I get this when I am trying to enter my create blog page. 

link to the page
```
<ul class="nav navbar-nav">
        {{#if isAuthenticated}}
      <li>{{#link-to "blog" class="navbar-brand"}}Create Blog{{/link-to}}</li>
        {{/if}}
      </ul>
``` 
blog.hbs
```
<h2>Create Blog</h2>

{{blog-form blog=model}}

{{outlet}}
```
blog-form
```
<div class="form-group">
  <label for="previous">Title</label>
  {{input type="text"
          class="form-control"
          maxlength="50"
          id="previous"
          placeholder="Title"
          value=passwords.previous}}
</div>

<div class="form-group">
  <label for="next">Subject</label>
  {{input type="text"
          class="form-control"
          maxlength="75"
          id="next"
          placeholder="Subject"
          value=passwords.next}}
</div>

<div class="form-group">
  <label for="next">Body</label>
  {{input type="text"
          class="form-control"
          maxlength="500"
          id="next"
          placeholder="Body"
          value=passwords.next}}
</div>

<div class="form-group">
  <label for="next">Author</label>
  {{input type="text"
          class="form-control"
          maxlength="50"
          id="next"
          placeholder="Author"
          value=passwords.next}}
</div>

<button type="submit" class="btn btn-primary" {{action "submit"}}>
  Create
</button>

<button class="btn btn-default" {{action "reset"}}>
  Cancel
</button>
```
The error the console is throwing doesn't give me a file to reference and I was trying to look in the ember tool and check the route, but I don't even think it's getting that far. 

I am also having a really hard time wrapping my head around organizing the routes in ember. Which I am will to and can open another issue about, but I think that is related to solving this problem. I believe this problem is a result of trying to define the actions for a user to be able to perform crud actions from the front end. The crud actions are working on the back end and whatever I make on the back end is able to display on the front end so that's cool. 

I have been trying to read the ember documentation, but I am just having a hard time making the connection, if there is anytime I would love the be able to talk a little about ember, I feel like I am really close to fixing both issues and that fixing one will help me understand and fix the other.

**Comment by raq929** [comment]
Do you have a diagram like we made in `ember-resources`? The first step is to draw one of those.

**Comment by novakw267** [comment]
I am not sure if this link will work, or if that is what you meant? 

https://www.draw.io/?lightbox=1&highlight=0000ff&edit=_blank&layers=1&nav=1&title=Untitled%20Diagram.xml#Uhttps%3A%2F%2Fraw.githubusercontent.com%2Fnovakw267%2FCapstone-frontend%2Freadme%2FUntitled%2520Diagram.xml

**Comment by novakw267** [comment]
I figured out why it was saying I was using an id already in use, originally I thought it wasn't generating a new Blog id but really it was the ID, FOR, and Values in the blog forms.

Since I had copied the forum structure from the change password form fields it didn't dawn on my to change those, or that those even existed.  
```
<div class="form-group">
  <label f**or="previous**">Title</label>
  {{input type="text"
          class="form-control"
          maxlength="50"
          **id="previous"**
          placeholder="Title"
          **value=passwords.previous**}}
</div>

<div class="form-group">
  <label **for="next"**>Subject</label>
  {{input type="text"
          class="form-control"
          maxlength="75"
          **id="next"**
          placeholder="Subject"
          **value=passwords.next**}}
</div>

<div class="form-group">
  <label **for="next"**>Body</label>
  {{input type="text"
          class="form-control"
          maxlength="500"
          **id="next"**
          placeholder="Body"
          **value=passwords.next**}}
</div>

<div class="form-group">
  <label **for="next"**>Author</label>
  {{input type="text"
          class="form-control"
          maxlength="50"
          **id="next"**
          placeholder="Author"
          **value=passwords.next**}}
</div>

<button type="submit" class="btn btn-primary" {{action "submit"}}>
  Create
</button>

<button class="btn btn-default" {{action "reset"}}>
  Cancel
</button>
```

After removing those it allowed the page to load the forms correctly.

TLDR: ID FOR and VALUE were defined and being used in the changepassword forms. So I had to edit or delete them. Be careful about copy pasting code, always make sure there are not single use pieces in it, and that you change all the things necessary.
